analyses/molecular-subtyping-EPN/02_ependymoma_generate_all_data.R

```diff
@@ -169,15 +198,42 @@ EPN_notebook[,"breaks_density-chromosomal_instability_SV"] <- sapply(EPN_noteboo
 
 # Adding focal CN from GISTIC files for CDKN2A
 EPN_notebook[,"GISTIC_focal_CN_CDKN2A"] <- sapply(EPN_notebook$Kids_First_Biospecimen_ID_DNA, 
-                                                  function(x) fill_df(sample = x, ref_df = gistic_focalCN, col_name = "CDKN2A", included_samples = cn_called_samples))
+                                                  function(x) fill_df(sample = x, ref_df = gistic_focalCN_CDKN2A, col_name = "CDKN2A", included_samples = cn_called_samples))
+EPN_notebook[,"GISTIC_focal_CN_MYCN"] <- sapply(EPN_notebook$Kids_First_Biospecimen_ID_DNA, 
+                                                  function(x) fill_df(sample = x, ref_df = gistic_focalCN_MYCN, col_name = "MYCN", included_samples = cn_called_samples))
+EPN_notebook[,"GISTIC_focal_CN_NF2"] <- sapply(EPN_notebook$Kids_First_Biospecimen_ID_DNA, 
+                                                function(x) fill_df(sample = x, ref_df = gistic_focalCN_NF2, col_name = "NF2", included_samples = cn_called_samples))
 
 # Adding focal CN from CDKN2A from CNV consensus files in analyses
 # Using status column from consensus_seg_annotated_cn_autosomes.tsv.gz file
-EPN_notebook[,"consensus_focal_CN_CDKN2"] <- sapply(EPN_notebook$Kids_First_Biospecimen_ID_DNA, 
+EPN_notebook[,"consensus_focal_CN_CDKN2A"] <- sapply(EPN_notebook$Kids_First_Biospecimen_ID_DNA, 
                                                     function(x) fill_df(sample = x, ref_df = focal_cn_gene_CDKN2A, col_name = "status", included_samples = cn_called_samples))
+EPN_notebook[,"consensus_focal_CN_MYCN"] <- sapply(EPN_notebook$Kids_First_Biospecimen_ID_DNA, 
+                                                    function(x) fill_df(sample = x, ref_df = focal_cn_gene_MYCN, col_name = "status", included_samples = cn_called_samples))
+EPN_notebook[,"consensus_focal_CN_NF2"] <- sapply(EPN_notebook$Kids_First_Biospecimen_ID_DNA, 
+                                                   function(x) fill_df(sample = x, ref_df = focal_cn_gene_NF2, col_name = "status", included_samples = cn_called_samples))
+
+mutation_fill_df <- function(sample, ref_df, col_name, default = 0){
+  if(is.na(sample)) { # no results for this sample
+    return("NA")
+  } else if(!sample %in% ref_df$Tumor_Sample_Barcode){ # sample is not present in the reference data set, return default
+    return(default)
+  } else {
+    value <- ref_df[ref_df$Tumor_Sample_Barcode == sample, col_name]
+    return(value)
+  }
+}
+
+
+EPN_notebook[,"NF2_variant_HGVSc"] <- unlist(sapply(EPN_notebook$Kids_First_Biospecimen_ID_DNA,
+                                          function(x) mutation_fill_df(sample = x, ref_df = mutations, col_name = 'HGVSc')))
+EPN_notebook[,"NF2_variant_consequence"] <- unlist(sapply(EPN_notebook$Kids_First_Biospecimen_ID_DNA,
```

Comment:
@rjcorb Are all three NF2 mutation columns required for subtyping or should there just be one relevant one?

Reply:
@atlas4213 these were the columns that @jharenza and I agreed to include in the output file. These columns provide unique and relevant information on detected NF2 variants, although they are not explicitly used for subtyping currently.